docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
 
 ### Auto interface implementation
 
-TODO: https://github.com/dotnet/roslyn/issues/72149
+**User scenario:** As a generator author I want to be able to implement the properties of the interfaces decorated with a specific attribute automatically for a user.
+
+**Solution:** Require the user to decorate the interface that they want to self-implement with the `[AutoImplement]` Attribute; Require the user to make the classes that implement the decorated interface be a `partial class`.
+Provide that attribute in a `RegisterPostInitializationOutput` step. Register for callbacks on the classes with
+`CreateSyntaxProvider` using the predicate `node is ClassDeclarationSyntax classDeclarationSyntax && classDeclarationSyntax.BaseList is not null && classDeclarationSyntax.BaseList.Types.Count > 0` to collect the classes that possibly implements at least one interface, And use tuples (or create an equatable model) to pass along that information.
+
+**Example:**
+
+```csharp
+public interface IUserInterface
+{
+    public int InterfaceProperty { get; set; }
+}
+
+public interface IUserInterface2
+{
+    public float InterfacePropertyOnlyGetter { get; }
+}
+```
+
+```csharp
+[AutoImplement(nameof(IUserInterface), nameof(IUserInterface2))]
+public partial class UserClass
+{
+    public string UserProp { get; set; }
+}
+```
+
+```csharp
+#nullable enable
+[Generator]
+public class AutoImplementGenerator : IIncrementalGenerator
+{
+    private const string NameOfString = "nameof(";
+    private const string AutoImplementAttributeNameSpace = "AttributeGenerator";
+    private const string AutoImplementAttributeName = "AutoImplement";
+    private const string AutoImplementAttributeClassName = $"{AutoImplementAttributeName}Attribute";
+    private const string FullyQualifiedMetadataName = $"{AutoImplementAttributeNameSpace}.{AutoImplementAttributeClassName}";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(ctx =>
+        {
+            //Generate the AutoImplement Attribute
+            const string autoImplementAttributeDeclarationCode = $$"""
+using System;
+namespace {{AutoImplementAttributeNameSpace}};
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
```

Comment:
Structs too? Fine if you want to say that's too complex for the workbook sample, but in that case, it would be good to call out the limitation in a comment explicitly.